ui/apps/everest/src/pages/settings/monitoring-endpoints/createEditModal/create-edit-modal.types.ts

```diff
@@ -18,13 +19,33 @@ export interface CreateEditEndpointModalProps {
   isLoading?: boolean;
 }
 
-export const endpointSchema = z.object({
-  [EndpointFormFields.name]: rfc_123_schema('endpoint name'),
-  [EndpointFormFields.namespaces]: z.array(z.string()).nonempty(),
-  [EndpointFormFields.url]: z.string().min(1).url(),
-  [EndpointFormFields.user]: z.string().min(1),
-  [EndpointFormFields.password]: z.string().min(1),
-});
+export const getEndpointSchema = (isEditMode: boolean) =>
+  z
+    .object({
+      [EndpointFormFields.name]: rfc_123_schema('endpoint name'),
+      [EndpointFormFields.namespaces]: z.array(z.string()).nonempty(),
+      [EndpointFormFields.url]: z.string().min(1).url(),
+      ...(isEditMode
```

Comment:
because we need to allow user submit form with 2 empty fields and in this case min(1) would be a problem. Users can submit form with 2 empty fields, becouse BE already has an api key, but in case when user wants to change api key to new, he mush fill both